File: src/main/java/com/tiobe/plugins/intellij/console/TICSConsole.java
package com.tiobe.plugins.intellij.console;

import com.intellij.execution.ExecutionManager;
import com.intellij.execution.executors.DefaultRunExecutor;
import com.intellij.execution.filters.TextConsoleBuilderFactory;
import com.intellij.execution.ui.ConsoleView;
import com.intellij.execution.ui.RunContentDescriptor;
import com.intellij.openapi.project.Project;
import com.intellij.execution.filters.Filter;

import javax.swing.JPanel;
import java.awt.BorderLayout;

public class TICSConsole {
    public static ConsoleView openConsole(final Project project, final String name, final Filter... filters) {
        final ConsoleView consoleView =
                TextConsoleBuilderFactory.getInstance().createBuilder(project).getConsole();
        for (final Filter filter : filters) {
            consoleView.addMessageFilter(filter);
        }

        final JPanel consolePanel = new JPanel(new BorderLayout());
        consolePanel.add(consoleView.getComponent(), BorderLayout.CENTER);

        final RunContentDescriptor runContentDescriptor = new RunContentDescriptor(consoleView, null, consolePanel, name) {
            public boolean isContentReuseProhibited() {
                return true; // prevents previously opened console from being reused
            }
        };
        ExecutionManager.getInstance(project).getContentManager().showRunContent(DefaultRunExecutor.getRunExecutorInstance(), runContentDescriptor);

        return consoleView;
    }
}

File: src/main/kotlin/com/tiobe/plugins/intellij/console/TicsConsole.kt
package com.tiobe.plugins.intellij.console

import com.intellij.execution.filters.TextConsoleBuilderFactory
import com.intellij.execution.process.ProcessHandler
import com.intellij.execution.ui.ConsoleView
import com.intellij.execution.ui.ConsoleViewContentType
import com.intellij.openapi.Disposable
import com.intellij.openapi.components.Service
import com.intellij.openapi.project.Project
import javax.swing.JComponent

@Service(Service.Level.PROJECT)
class TicsConsole(project: Project): Disposable {
    private var consoleView: ConsoleView = TextConsoleBuilderFactory.getInstance().createBuilder(project).console

    fun attachToProcess(handler: ProcessHandler, firstLine: String? = null) {
        consoleView.clear()
        firstLine?.let {
            consoleView.print("$firstLine\n", ConsoleViewContentType.SYSTEM_OUTPUT)
        }
        consoleView.attachToProcess(handler)
    }

    fun clear() {
        consoleView.clear()
    }

    fun getComponent(): JComponent {
        return consoleView.component
    }

    override fun dispose() {
        consoleView.dispose()
    }
}